thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js

```diff
@@ -0,0 +1,256 @@
+   * @method renderDate
+   * @param {Number} range - number of days (duration)
+   * @return {String}
+   */
+  renderDate: function(range) {
+    const newDate = moment().subtract(range, 'days').endOf('day').utc().format("DD MM YYY");
+    return `Last ${range} Days (${newDate} to Today)`;
+  },
+
+  /**
+   * Array containing the types of anomaly responses supported
+   * @type {Array}
+   */
+  responseOptions: ['Not reviewed yet', 'True anomaly', 'False alarm', 'I don\'t know', 'Confirmed - New Trend'],
+
+  /**
+   * Data needed to render the stats 'cards' above the anomaly graph for this alert
+   * @type {Object}
+   */
+  anomalyStats: Ember.computed(
+    'totalAnomalies',
+    function() {
+      const total = this.get('totalAnomalies');
+      const anomalyStats = [
+        {
+          title: 'Number of anomalies',
+          text: 'Estimated average number of anomalies per month',
+          value: total,
+        },
+        {
+          title: 'Response rate',
+          text: 'Percentage of anomalies that has a response',
+          value: '87.1%'
+        },
+        {
+          title: 'Precision',
+          text: 'Among all anomalies detected, the percentage of them that are true.',
+          value: '50%'
+        },
+        {
+          title: 'Recall',
+          text: 'Among all anomalies that happened, the percentage of them detected by the system',
+          value: '25%'
+        },
+      ];
+      return anomalyStats;
+    }
+  ),
+
+  /**
+   * If user selects a dimension from the dropdown, we filter the anomaly results here.
+   * NOTE: this is currently set up to support single-dimension filters
+   * @type {Object}
+   */
+  filteredAnomalies: Ember.computed(
+    'selectedDimension',
+    'anomalyData',
+    function() {
+      const targetDimension = this.get('selectedDimension');
+      let anomalies = this.get('anomalyData');
+
+      if (targetDimension === 'All Dimensions') {
+        return anomalies;
+      } else {
+        return anomalies.filter(data => {
+          return targetDimension === `${data.dimensionList[0].dimensionKey}:${data.dimensionList[0].dimensionVal}`;
+        });
+      }
+    }
+  ),
+
+  /**
+   * Placeholder for options for range field. Here we generate arbitrary date ranges from our config.
+   * @type {Array}
+   */
+  rangeOptionsExample: Ember.computed(
+    'dateRangeToRender',
+    function() {
+      let dateRanges = [];
+      this.get('dateRangeToRender').forEach(date => {
+        dateRanges.push(this.renderDate(date))
+      });
+      return dateRanges;
+    }
+  ),
+
+  /**
+   * Pings the job-info endpoint to check status of an ongoing replay job.
+   * If there is no progress after a set time, we display an error message.
+   * TODO: Set error message on timeout
+   * @method checkReplayStatus
+   * @param {Number} jobId - the id for the newly triggered replay job
+   * @return {fetch promise}
+   */
+  checkReplayStatus(jobId) {
+    const that = this;
```

Comment:
let's stay away from saving `this` into another a value. 
The combination of `arrow functions` and passing this context in `Ember.run.later` should allow us to get rid of the `that` 
```
run.later(myContext, function() {
  // code here will execute within a RunLoop in about 500ms with this == myContext
}, 500);
```
https://emberjs.com/api/ember/2.15/namespaces/Ember.run/methods/later?anchor=later

Reply:
^ This the only blocking comment.

Reply:
Ah, yes - I forgot to change that!